sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -343,6 +343,35 @@ class DataFrame private[sql](
   }
 
   /**
+   * Inner equi-join with another [[DataFrame]] using the column.
+   * {{{
+   *   // Joining df1 and df2 using the column "user_id"
+   *   df1.join(df2, "user_id")
+   * }}}
+   *
+   * Note that if you perform a self-join using this function, you wouldn't be able to reference
```

Comment:
... if you perform a self-join _without aliasing the input DataFrames_ ...
nit: you _will not`_ be able ?